kafka_checked/src/main.rs

```diff
@@ -54,72 +55,125 @@ struct Tbtime {
     #[serde(rename = "@val")]
     val: String, // avoiding parsing time at this point...
     sensor: Vec<Sensor>,
-    kvtextdata: Option<Vec<Kvtextdata>>,
+    _kvtextdata: Option<Vec<Kvtextdata>>,
 }
 #[derive(Debug, Deserialize)]
 /// Represents <sensor>...</sensor>
 struct Sensor {
     #[serde(rename = "@val")]
-    val: i32,
+    val: Option<i32>,
     level: Vec<Level>,
 }
 /// Represents <level>...</level>
 #[derive(Debug, Deserialize)]
 struct Level {
     #[serde(rename = "@val")]
-    val: i32,
+    val: Option<i32>,
     kvdata: Option<Vec<Kvdata>>,
```

Comment:
I just noticed this weird hierarchy we have with level inside sensor, and kvdata inside sensor that doesn't make much intuitive sense. I get that that's how the xml is structured, but I'm not sure how intentional that structure is, or whether they just nested things because that seemed natural in xml.

I think it might make sense to use serde flattening, do ditch the weird hierarchy and combine the structs into one?

https://serde.rs/attr-flatten.html

Reply:
I agree 100% that this nested structure does not look ideal, however as far as I could see by skimming through some of the messages in the queue, all tags (except maybe `station` and `typeid`, but I wouldn't exclude it) can indeed "randomly" contain sequences of subtags, so it might be difficult to implement flattening here.